### PR TITLE
Replace Dotnet.Cli.Utils.Command with System.Diagnostic.Process

### LINKED ISSUE #92: jit-diff should show path of base and diff
To allow for easy cut-and-paste to a directory diff program. e.g.,:
```
To examine diffs, use a command like:
   <diff-program> c:\diffs\base\4 c:\diffs\diff\4

e.g., on Windows:
    windiff c:\diffs\base\4 c:\diffs\diff\4
```

On Windows, our desktop testing infrastructure also automatically detects if the paths are shared, giving a UNC path. This is useful for copy-and-paste of the command to email to others to look at your diffs, e.g.:
```
   windiff \\brucefo1\mydiffs\base\4 \\brucefo1\mydiffs\diff\4
```

### LINKED ISSUE #141: jit-diff and jit-diff-pmi verbose output should include exact set of environment variable settings
When trying to reproduce a failure of the tools to run, using the `--verbose` switch should give all the information you need. This should include all of the `COMPlus` and other environment variables that are set before a command is run, including their values.

Currently, only some "ambient" variable values that are passed through are output, plus the not-very-useful "Setting AltJit" message.

The output should be in a form that can by copy-pasted on the platform, e.g., `set COMPlus_JitDisasm=*`, etc.

### LINKED ISSUE #151: jit-diff: implement better cancellation mechanism
Seems like it takes multiple ^C and a long time to bring jit-diffs to a stop. When you kick one of these off with big assembly lists and things go very wrong right away it should be easier to cancel the whole thing.

I had some prototypes where I implemented the ^C handler and passed cancellation tokens around in jit-diffs, but given the interprocess way this all plays out we likely also need to plumb cancellation events down into the guts of jit-dasm and pmi. Upper layers can pass the event names down to lower layers as optional args or something.

### LINKED ISSUE #175: jit-diff eats jit-analyze newlines
When jit-diff invokes jit-analyze, it appears to eat all the solo newlines, eliminating all blank lines. When jit-analyze is run alone, these blank lines are there.

E.g., from jit-diff:
```
Diffs (if any) can be viewed by comparing: f:\diffs\dasmset_32\base f:\diffs\dasmset_32\diff
Analyzing diffs...
Found 2 files with textual diffs.
PMI Diffs for System.Private.CoreLib.dll for x64 protononjit.dll
Summary:
(Lower is better)
Total bytes of diff: 1224 (0.21% of base)
    diff is a regression.
Top file regressions by size (bytes):
        1224 : System.Private.CoreLib.dasm (0.21% of base)
1 total files with size differences (0 improved, 1 regressed), 0 unchanged.
Top method regressions by size (bytes):
         188 ( 2.74% of base) : System.Private.CoreLib.dasm - System.DefaultBinder:BindToMethod(int,ref,byref,ref,ref,ref,byref):ref:this
          24 ( 2.45% of base) : System.Private.CoreLib.dasm - Sha1ForNonSecretPurposes:Drain():this
          24 ( 1.27% of base) : System.Private.CoreLib.dasm - System.Enum:TryParseEnum(ref,ref,bool,byref):bool
          24 ( 0.39% of base) : System.Private.CoreLib.dasm - System.RuntimeType:InvokeMember(ref,int,ref,ref,ref,ref,ref,ref):ref:this
          20 ( 1.68% of base) : System.Private.CoreLib.dasm - System.Diagnostics.Tracing.EventSource:InitializeProviderMetadata():this
Top method improvements by size (bytes):
         -40 (-2.69% of base) : System.Private.CoreLib.dasm - System.AppDomain:Setup(ref):ref
         -12 (-3.61% of base) : System.Private.CoreLib.dasm - SorterObjectArray:SwapIfGreaterWithItems(int,int):this
         -12 (-5.00% of base) : System.Private.CoreLib.dasm - SorterObjectArray:Swap(int,int):this
          -8 (-1.72% of base) : System.Private.CoreLib.dasm - System.AppDomain:PrepareDataForSetup(ref,ref,ref,ref):ref
          -8 (-0.60% of base) : System.Private.CoreLib.dasm - System.Collections.Generic.Dictionary`2[__Canon,__Canon][System.__Canon,System.__Canon]:TryInsert(ref,ref,ubyte):bool:this
Top method regressions by size (percentage):
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):ref:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):int:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):double:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):long:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:set_Item(int,int):this
Top method improvements by size (percentage):
         -12 (-5.00% of base) : System.Private.CoreLib.dasm - SorterObjectArray:Swap(int,int):this
         -12 (-3.61% of base) : System.Private.CoreLib.dasm - SorterObjectArray:SwapIfGreaterWithItems(int,int):this
         -40 (-2.69% of base) : System.Private.CoreLib.dasm - System.AppDomain:Setup(ref):ref
          -8 (-1.72% of base) : System.Private.CoreLib.dasm - System.AppDomain:PrepareDataForSetup(ref,ref,ref,ref):ref
          -8 (-1.37% of base) : System.Private.CoreLib.dasm - System.Reflection.MethodBase:CheckArguments(ref,ref,int,ref,ref):ref:this
173 total methods with size differences (12 improved, 161 regressed), 2277 unchanged.
Completed analysis in 0.82s
```

from jit-analyze:
```
Found 2 files with textual diffs.

Summary:
(Lower is better)

Total bytes of diff: 1224 (0.21% of base)
    diff is a regression.

Top file regressions by size (bytes):
        1224 : System.Private.CoreLib.dasm (0.21% of base)

1 total files with size differences (0 improved, 1 regressed), 0 unchanged.

Top method regressions by size (bytes):
         188 ( 2.74% of base) : System.Private.CoreLib.dasm - System.DefaultBinder:BindToMethod(int,ref,byref,ref,ref,ref,byref):ref:this
          24 ( 2.45% of base) : System.Private.CoreLib.dasm - Sha1ForNonSecretPurposes:Drain():this
          24 ( 1.27% of base) : System.Private.CoreLib.dasm - System.Enum:TryParseEnum(ref,ref,bool,byref):bool
          24 ( 0.39% of base) : System.Private.CoreLib.dasm - System.RuntimeType:InvokeMember(ref,int,ref,ref,ref,ref,ref,ref):ref:this
          20 ( 1.68% of base) : System.Private.CoreLib.dasm - System.Diagnostics.Tracing.EventSource:InitializeProviderMetadata():this

Top method improvements by size (bytes):
         -40 (-2.69% of base) : System.Private.CoreLib.dasm - System.AppDomain:Setup(ref):ref
         -12 (-3.61% of base) : System.Private.CoreLib.dasm - SorterObjectArray:SwapIfGreaterWithItems(int,int):this
         -12 (-5.00% of base) : System.Private.CoreLib.dasm - SorterObjectArray:Swap(int,int):this
          -8 (-1.72% of base) : System.Private.CoreLib.dasm - System.AppDomain:PrepareDataForSetup(ref,ref,ref,ref):ref
          -8 (-0.60% of base) : System.Private.CoreLib.dasm - System.Collections.Generic.Dictionary`2[__Canon,__Canon][System.__Canon,System.__Canon]:TryInsert(ref,ref,ubyte):bool:this

Top method regressions by size (percentage):
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):ref:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):int:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):double:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:get_Item(int):long:this
           4 ( 7.14% of base) : System.Private.CoreLib.dasm - System.SZArrayHelper:set_Item(int,int):this

Top method improvements by size (percentage):
         -12 (-5.00% of base) : System.Private.CoreLib.dasm - SorterObjectArray:Swap(int,int):this
         -12 (-3.61% of base) : System.Private.CoreLib.dasm - SorterObjectArray:SwapIfGreaterWithItems(int,int):this
         -40 (-2.69% of base) : System.Private.CoreLib.dasm - System.AppDomain:Setup(ref):ref
          -8 (-1.72% of base) : System.Private.CoreLib.dasm - System.AppDomain:PrepareDataForSetup(ref,ref,ref,ref):ref
          -8 (-1.37% of base) : System.Private.CoreLib.dasm - System.Reflection.MethodBase:CheckArguments(ref,ref,int,ref,ref):ref:this

173 total methods with size differences (12 improved, 161 regressed), 2277 unchanged.
```

This seems to have something to do with util.cs Utility.TryCommand using Microsoft.DotNet.Cli.Utils, in particular ForwardStdOut().

### PR DESCRIPTION
**Pull request by kunalspathak**
- Replaced `Dotnet.Cli.Utils.Command` with `System.Diagnostic.Process` which is more robust and allow us to better handle process spawning mechanism.
- Print the `git diff` command to fix #92 
- Switch to `System.Diagnostic.Process` to fix #175.
- Add cancellation mechanism that will kill all the process and child process the given jitutil has spawned. Fixes #151 .
- Print the environment variables that fixes #141.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by kunalspathak** [comment]
Can someone take a look into this?

File: src/jit-dasm/jit-dasm.cs [hunk 1]
```diff
@@ -518,41 +512,36 @@ public void GenerateAsm()
...
                         var logPath = Path.ChangeExtension(dasmPath, ".log");
+                        result = ExecuteProcess(commandArgs);
 
-                        // Redirect stdout/stderr to log file and run command.
-                        using (var outputStreamWriter = File.CreateText(logPath))
+                        // Write stdout/stderr to log file.
+                        StringBuilder output = new StringBuilder();
+                        if (!string.IsNullOrEmpty(result.StdOut))
+                        {
+                            output.AppendLine(result.StdOut);
```
Comment (AndyAyersMS) [inline src/jit-dasm/jit-dasm.cs:536]: Despite the `AppendLine` this is potentially mutli-line output?

Hoping this doesn't lose or encode line breaks like we see in CI results sometimes.

**Review by AndyAyersMS — APPROVED** [review]
Looks good, just one small question.

After you merge this, keep an eye on subsequent CI jobs, because they'll pick up your formatter changes.

**Review by BruceForstall — APPROVED** [review]
LGTM

File: src/jit-dasm/jit-dasm.cs (hunk 1 above)
Comment (kunalspathak) [reply src/jit-dasm/jit-dasm.cs:536]: Which CI job are you referring? By the way, thanks for pointing this. It had bug. It should have passed `capture=true` to `ExecuteProcess()` so we get back `result.StdOut`. Also, I have changed to `output.Append()` without adding extra line.

Comment (AndyAyersMS) [reply src/jit-dasm/jit-dasm.cs]: One of those where we see lots of long lines with embedded `\r\n` for the captured output, eg:

image

Comment (kunalspathak) [reply src/jit-dasm/jit-dasm.cs]: I see. Sure, I will check it out periodically.

**Pull request merged**